### PR TITLE
migrator: faster implementation of count_citations

### PR DESCRIPTION
**Pull request by jacquerie**
Closes #1495

### CODE REVIEW

File: inspirehep/modules/migrator/tasks.py [hunk 1]
```diff
@@ -237,58 +241,47 @@ def get_records_to_update_generator(citation_lookup):
...
+        _get_records_to_update_generator(citations_lookup),
         chunk_size=chunk_size,
         raise_on_exception=True,
         raise_on_error=True,
         request_timeout=request_timeout,
-        stats_only=True)
-    click.echo("DONE: {} records updated with success. {} failures.".format(
+        stats_only=True,
+    )
+    click.echo('... DONE: {} records updated with success. {} failures.'.format(
```
Comment (kaplun) [inline inspirehep/modules/migrator/tasks.py]: So single quotes are better than double?

File: inspirehep/modules/migrator/tasks.py [hunk 2]
```diff
@@ -237,58 +241,47 @@ def get_records_to_update_generator(citation_lookup):
...
-                for reference in references:
-                    recid = reference.get('recid')
-                    if recid:
-                        if isinstance(recid, list):
-                            # Sometimes there is more than one recid in the
-                            # reference.
-                            recid = recid.pop()
-                        unique_refs_ids.add(recid)
+            unique_refs_ids = dedupe_list(list(chain.from_iterable(map(
+                force_force_list, get_value(record, '_source.references.recid')))))
```
Comment (kaplun) [inline inspirehep/modules/migrator/tasks.py]: Mmh I am not really understanding how you have refactored, but I trust is working :+1:

Comment (jacquerie) [inline inspirehep/modules/migrator/tasks.py]: http://imgur.com/iZcUNxH

Comment (kaplun) [inline inspirehep/modules/migrator/tasks.py]: Adesso sono turbato

Comment (kaplun) [inline inspirehep/modules/migrator/tasks.py]: Can't sleep anylonger

**Pull request merged**